src/Microsoft.AspNet.Razor/CodeGenerators/CodeWriter.cs

```diff
@@ -10,32 +9,21 @@ namespace Microsoft.AspNet.Razor.CodeGenerators
     public class CodeWriter : IDisposable
     {
         private static readonly char[] NewLineCharacters = new char[] { '\r', '\n' };
-        private readonly StringWriter _writer = new StringWriter();
-        private bool _newLine;
+
         private string _cache = string.Empty;
-        private bool _dirty = false;
+        private bool _dirty;
 
         private int _absoluteIndex;
         private int _currentLineIndex;
         private int _currentLineCharacterIndex;
 
-        public StringBuilder Builder => _writer.GetStringBuilder();
-
-        public string LastWrite { get; private set; }
+        public StringBuilder Builder { get; } = new StringBuilder();
 
         public int CurrentIndent { get; private set; }
 
-        public string NewLine
-        {
-            get
-            {
-                return _writer.NewLine;
-            }
-            set
-            {
-                _writer.NewLine = value;
-            }
-        }
+        public bool IsAfterNewLine { get; private set; }
+
+        public string NewLine { get; set; } = Environment.NewLine;
```

Comment:
> Yes, the test infrastructure changes this accordingly.

If the test infrastructure ensures that mixes real people will hit don't occur when testing, that's bad. If you mean the test infrastructure mixes line endings to confirm they're correctly handled, that's good. So which did you mean?